csrc/flash_attn/flash_api.cpp

```diff
@@ -159,6 +168,7 @@ void run_mha_fwd(Flash_fwd_params &params, cudaStream_t stream, bool force_split
         HEADDIM_SWITCH(params.d, [&] {
             BOOL_SWITCH(params.is_causal, Is_causal, [&] {
                 if (params.num_splits <= 1 && !force_split_kernel) {  // If we don't set it num_splits == 0
+                    assert(false);
```

Comment:
What is going on here? Perhaps you mean to also remove the call that follows?

Reply:
Good catch. That was just to make sure it was launching the right kernel when I was testing. It should be removed